ntp-daemon/src/config/peer.rs

```diff
@@ -47,34 +43,55 @@ impl PeerConfig {
     }
 }
 
-fn normalize_addr(mut addr: String) -> std::io::Result<String> {
-    if addr.split(':').count() > 2 {
-        // IPv6, try to parse it as such
-        match addr.parse::<SocketAddr>() {
-            Ok(_) => Ok(addr),
-            Err(e) => {
-                // Could be because of no port, add one and see
-                addr = format!("[{addr}]:123");
-                if addr.parse::<SocketAddr>().is_ok() {
-                    Ok(addr)
-                } else {
-                    Err(std::io::Error::new(std::io::ErrorKind::Other, e))
+/// A normalized address has a host and a port part. However, the host may be
+/// invalid, we didn't yet perform a DNS lookup.
+#[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct NormalizedAddress(String);
```

Comment:
Since this purports to ensure some properties are checked, I think we should use a private field here (and hence not a tuple struct).